llvm/lib/Target/SPIRV/SPIRVStructurizer.cpp

```diff
@@ -447,55 +439,51 @@ class SPIRVStructurizer : public FunctionPass {
     // clang-format on
     std::vector<Edge>
     createAliasBlocksForComplexEdges(std::vector<Edge> Edges) {
-      std::unordered_map<BasicBlock *, BasicBlock *> Seen;
+      std::unordered_set<BasicBlock *> Seen;
       std::vector<Edge> Output;
       Output.reserve(Edges.size());
 
       for (auto &[Src, Dst] : Edges) {
-        auto [iterator, inserted] = Seen.insert({Src, Dst});
-        if (inserted) {
-          Output.emplace_back(Src, Dst);
-          continue;
+        auto [iterator, inserted] = Seen.insert(Src);
+        if (!inserted) {
+          // Src already a source node. Cannot have 2 edges from A to B.
+          // Creating alias source block.
+          BasicBlock *NewSrc =
+              BasicBlock::Create(F.getContext(), "new.src", &F);
```

Comment:
Not sure about this random hard coded name for BB. I'd understand it either empty, or basing on the `Src` name (as an option, Src->printAsOperand()), or as a prefix with numbering, or anything except for the hard coded and the same for all transformation cases.
Please feel to ignore this if you see why the current version is better, because I can imagine a case when "new.src" makes sense.

Reply:
The name is automatically suffixed with a number if another block is already named "new.src".
I wanted to be able to know where a block came from when reading LLVM-IR, but also something simple.
Something I can also do is to prefix the name with the block it relates to. Simple, and helps even more.